content/news/2021-08-19-gsoc-coverage.md

```diff
@@ -0,0 +1,130 @@
+sharing part was possible to implement with the current macros available in
+Ruma, storage requires a major refactoring of these macros. This means that
+although I was able to do the groundwork for supporting [MSC1946], secret
+storage will not be usable until those macros are reworked. 
+
+The reason for all this trouble is that [MSC1946] introduces an event
+`*m.secret_storage.profile.[key_ID]*` where `[key_ID]` is a placeholder for the
+ID of a key to be stored. This means there could be any number of event types
+possible, instead of the until now predefined set of event types. I thus opened
+a [PR] for this to be supported in Ruma. 
+
+Although secret storage will need to wait to be merged, secret sharing has
+already been merged. This part of [MSC1946], as its name suggests, allows
+clients to share secrets between devices. As such, keys can be exchanged between
+devices using the  part of the Matrix Spec introduced by this MSC. In fact,
+[`matrix-rust-sdk`] has already implemented [the gossiping of cross signing
+keys], building on top of Ruma's implementation of Secret Sharing.
+
+As I wasn't realistic to include the major macro refactor in my GSoC project, I
+decided to move onto  [MSC2785: Event notification attributes and actions]. I
+had kept this as an extra goal for my project, and seemed to be a good fit for
+the situation. Unluckily though, when I read more into the MSC's text, I
+realized I would encounter the exact same problem as I had with Secret Storage.
+This MSC introduced the event type `*m.notifications_profile.<profile>*` where
+`<profile>` is the name of the profile to store. I still powered ahead,
+implementing the different endpoints and events, temporarily setting a fixed
+event type name for `*m.notifications_profile.<profile>`.  This allowed the
+grunt of the work to be done, meaning it would be ready for when Ruma gets
+support for variable event names. Although the MSC isn't finalized as of yet,
+all the currently laid out endpoints and events have been implemented in a
+branch ready to be merged once possible.
+
+Thus there are a couple of pull requests which are blocked and that will be
+merged in the future, once variable event type names are supported:
+
+- [Add support for MSC2785: Event notification attributes and actions]
+- [Add secret storage]
+
+Finally, I worked on a final issue for my GSoC project:
+
+- [Generate for impls for event enums]
+
+To solve this last issue, I decided to learn how to write procedural macros. In
+the first half of GSoC, I had created a new `macro_rules!` macro, largely based
+on a pre-existing one. As I had very little experience in writing macros, and
+zero experience in writing procedural ones, I decided to read a short series of
+[blog posts] to get myself up to speed. Once that was done, I got to work: I
+added the generation of `From` trait implementations inside the pre-existing
+`[proc-macro]` functions which generate the event enums. I also added a `Derive`
```

Comment:
Maybe don't highlight `[proc-macro]` like it was some sort of syntax within Rust and just have sth. like

```suggestion
proc-macro code which generate the event enums. I also added a `Derive`
```

Reply:
Same for "Derive", really. I don't think it needs to be captialized or highlighted.